### PR TITLE
ref(alert rule): Check for rule existing before adding value

### PR DESCRIPTION
**Pull request by ceorourke**
I can't figure out _how_ this happens, but checking that `alert_rule` exists before trying to add a value to a non-existent dict should resolve the issue.

Fixes https://sentry.sentry.io/issues/5245194605/

### CODE REVIEW

**Review by nhsiehgit — APPROVED** [review]
can we add a test?

**Review by nhsiehgit — CHANGES_REQUESTED** [review]
blocking merge

pushed up some changes to better address

File: src/sentry/api/serializers/models/alert_rule.py [hunk 1]
```diff
@@ -330,20 +333,33 @@ def get_attrs(
...
         serialized_alert_rules = serialize(alert_rules, user=user)
-        rules = serialize(
+        serialized_map_by_id = {x.id: x for x in serialized_alert_rules}
+
+        serialized_issue_rules = serialize(
             [x for x in item_list if isinstance(x, Rule)],
             user=user,
             serializer=RuleSerializer(expand=self.expand),
         )
+        serialized_issue_rule_map_by_id = {x.id for x in serialized_issue_rules}
```
Comment (saponifi3d) [inline src/sentry/api/serializers/models/alert_rule.py:343]: Nit: just get the keys of the data structure from earlier
```suggestion
        serialized_issue_rule_ids = serialized_map_by_id.keys()
```
--
Do we need to create this variable or could we just use the map?

File: src/sentry/api/serializers/models/alert_rule.py [hunk 2]
```diff
@@ -330,20 +333,33 @@ def get_attrs(
...
+            if item_id in serialized_map_by_id:
+                serialized_alert_rule = serialized_map_by_id[item_id]
+                if "latestIncident" in self.expand:
+                    serialized_alert_rule["latestIncident"] = incident_map.get(item.incident_id)  # type: ignore[attr-defined]
+                results[item] = serialized_alert_rule
+            elif item_id in serialized_issue_rule_map_by_id:
+                results[item] = serialized_issue_rule_map_by_id[item_id]
+            else:
+                logger.warning(
+                    "Alert Rule found but dropped during serialization",
```
Comment (saponifi3d) [inline src/sentry/api/serializers/models/alert_rule.py:356]: is there a case when a user creates an alert rule and it fails serialization, but it is created in the db? i'm a little nervous about if a user has an alert rule that can be triggered, but isn't visible via the API. If that's possible maybe we should ramp this up to a logger.error?

File: src/sentry/api/serializers/models/alert_rule.py (hunk 1 above)
Comment (nhsiehgit) [reply src/sentry/api/serializers/models/alert_rule.py:343]: Ah - the serialized alert rules are different than the serialized issue rules.

and yea - the map of id -> instance _is_ still needed to easily grab the instance

File: src/sentry/api/serializers/models/alert_rule.py (hunk 2 above)
Comment (nhsiehgit) [reply src/sentry/api/serializers/models/alert_rule.py:356]: Yea - i'll make this an error instead 👍

**Review by nhsiehgit — APPROVED** [review]

**Pull request merged**